### PR TITLE
Remove /signup routes

### PR DESCRIPTION
**Pull request by arku**
#### What? Why?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/6418

Removes `GET /signup` and `POST /signup` routes and the controller actions as they aren't needed anymore.

#### Release notes

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

File: config/routes/spree.rb [hunk 1]
```diff
@@ -15,7 +15,6 @@
   devise_scope :spree_user do
     post '/login' => 'user_sessions#create', :as => :create_new_session
     get '/logout' => 'user_sessions#destroy', :as => :logout
-    get '/signup' => 'user_registrations#new', :as => :signup
     post '/signup' => 'user_registrations#create', :as => :registration
```
Comment (arku) [inline config/routes/spree.rb:18]: Are we using this route anywhere?

**Comment by arku** [comment]
The failing spec on CI passes locally 😕

**Comment by sigmundpetersen** [comment]
Yes, these are flaky spec failures, don't worry.
I will rebuild to get it green.

**Review by andrewpbrett — APPROVED** [review]
Thanks @arku!

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
It'd be great to also check 

> the removal of app/controllers/spree/user_registrations_controller.rb.

We might need to create a separate issue out of https://github.com/openfoodfoundation/openfoodnetwork/issues/6418.

**Comment by Matt-Yorkley** [comment]
Shall we move this to test ready? It's got two approvals.

**Comment by filipefurtad0** [comment]
Hi @arku ,

Verified the error 500 before staging, by visiting `/signup` and observing the console:

image

And the bugsnag error, in staging-AU.

After staging this PR, this triggers error 404:

image

So, the error does not trigger Bugsnag anymore. I think this closes #6418 , moving to ready to go :+1: 

Regarding the removal of removal of `app/controllers/spree/user_registrations_controller.rb.`: Do we need an issue (tech-debt) for this?

**Comment by arku** [comment]
> Regarding the removal of removal of app/controllers/spree/user_registrations_controller.rb.: Do we need an issue (tech-debt) for this?

@filipefurtad0 Yeah.

**Comment by filipefurtad0** [comment]
Thanks for your feedback @arku, 
I opened tech-debt #6531.

**Pull request merged**